opamp/observiq/observiq_packages_state_provider_test.go

```diff
@@ -0,0 +1,499 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observiq
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/observiq/observiq-otel-collector/internal/version"
+	"github.com/open-telemetry/opamp-go/client/types"
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+func TestNewPackagesStateProvider(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		testFunc func(*testing.T)
+	}{
+		{
+			desc: "New PackagesStateProvider",
+			testFunc: func(*testing.T) {
```

Comment:
```suggestion
			testFunc: func(t *testing.T) {
```
I think this is a problem that's present in a lot (all?) of these tests; We should shadow the parent test, otherwise when it fails you get: 
```
testing.go:1169: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
```

Reply:
@BinaryFissionGames changed these